wdl-analysis/src/analyzer.rs

```diff
@@ -501,5 +600,256 @@ impl Drop for Analyzer {
+            .expect("should add document");
+
+        let results = analyzer.analyze(()).await;
+        assert_eq!(results.len(), 1);
+        assert_eq!(results[0].diagnostics().len(), 1);
+        assert_eq!(results[0].diagnostics()[0].rule(), None);
+        assert_eq!(results[0].diagnostics()[0].severity(), Severity::Error);
+        assert_eq!(
+            results[0].diagnostics()[0].message(),
+            "conflicting workflow name `test`"
+        );
+
+        // Analyze again and ensure the analysis result id is unchanged
+        let id = results[0].id().clone();
+        let results = analyzer.analyze(()).await;
+        assert_eq!(results.len(), 1);
+        assert_eq!(results[0].id().as_ref(), id.as_ref());
+        assert_eq!(results[0].diagnostics().len(), 1);
+        assert_eq!(results[0].diagnostics()[0].rule(), None);
+        assert_eq!(results[0].diagnostics()[0].severity(), Severity::Error);
+        assert_eq!(
+            results[0].diagnostics()[0].message(),
+            "conflicting workflow name `test`"
+        );
+    }
+
+    #[tokio::test]
+    async fn it_reanalyzes_a_document_on_change() {
+        let dir = TempDir::new().expect("failed to create temporary directory");
+        let path = dir.path().join("foo.wdl");
+        fs::write(
+            &path,
+            r#"version 1.1
+
+task test {
+    command <<<>>>
+}
+
+workflow test {
+}
+"#,
+        )
+        .expect("failed to create test file");
+
+        // Analyze the file and check the resulting diagnostic
+        let analyzer = Analyzer::new(|_: (), _, _, _| async {});
+        analyzer
+            .add_documents(vec![path.clone()])
+            .await
+            .expect("should add document");
+
+        let results = analyzer.analyze(()).await;
+        assert_eq!(results.len(), 1);
+        assert_eq!(results[0].diagnostics().len(), 1);
+        assert_eq!(results[0].diagnostics()[0].rule(), None);
+        assert_eq!(results[0].diagnostics()[0].severity(), Severity::Error);
+        assert_eq!(
+            results[0].diagnostics()[0].message(),
+            "conflicting workflow name `test`"
+        );
+
+        let uri = path_to_uri(&path).expect("should convert to URI");
+        analyzer.notify_change(uri.clone(), false);
+
+        // Analyze again and ensure the analysis result id is changed
+        let id = results[0].id().clone();
+        let results = analyzer.analyze(()).await;
+        assert_eq!(results.len(), 1);
+        assert!(results[0].id().as_ref() != id.as_ref());
+        assert_eq!(results[0].diagnostics().len(), 1);
+        assert_eq!(results[0].diagnostics()[0].rule(), None);
+        assert_eq!(results[0].diagnostics()[0].severity(), Severity::Error);
+        assert_eq!(
+            results[0].diagnostics()[0].message(),
+            "conflicting workflow name `test`"
+        );
+
+        // Analyze again and ensure the analysis result id is unchanged
+        let id = results[0].id().clone();
```

Comment:
Not required IMO, but it could be interesting to rewrite the file without the
conflicting name here and check that the diagnostic resolves itself (sort of an analog
to what you do with the incremental test).

Reply:
Added and this caught a bug!